helix-term/src/keymap.rs

```diff
@@ -62,252 +283,298 @@ impl DerefMut for Keymaps {
 
 impl Default for Keymaps {
     fn default() -> Keymaps {
-        let normal = hashmap!(
-            key!('h') => Command::move_char_left,
-            key!('j') => Command::move_line_down,
-            key!('k') => Command::move_line_up,
-            key!('l') => Command::move_char_right,
-
-            key!(Left) => Command::move_char_left,
-            key!(Down) => Command::move_line_down,
-            key!(Up) => Command::move_line_up,
-            key!(Right) => Command::move_char_right,
-
-            key!('t') => Command::find_till_char,
-            key!('f') => Command::find_next_char,
-            key!('T') => Command::till_prev_char,
-            key!('F') => Command::find_prev_char,
-            // and matching set for select mode (extend)
-            //
-            key!('r') => Command::replace,
-            key!('R') => Command::replace_with_yanked,
-
-            key!('~') => Command::switch_case,
-            alt!('`') => Command::switch_to_uppercase,
-            key!('`') => Command::switch_to_lowercase,
-
-            key!(Home) => Command::goto_line_start,
-            key!(End) => Command::goto_line_end,
-
-            key!('w') => Command::move_next_word_start,
-            key!('b') => Command::move_prev_word_start,
-            key!('e') => Command::move_next_word_end,
-
-            key!('W') => Command::move_next_long_word_start,
-            key!('B') => Command::move_prev_long_word_start,
-            key!('E') => Command::move_next_long_word_end,
-
-            key!('v') => Command::select_mode,
-            key!('g') => Command::goto_mode,
-            key!(':') => Command::command_mode,
-
-            key!('i') => Command::insert_mode,
-            key!('I') => Command::prepend_to_line,
-            key!('a') => Command::append_mode,
-            key!('A') => Command::append_to_line,
-            key!('o') => Command::open_below,
-            key!('O') => Command::open_above,
+        let normal = keymap!({ "Normal mode"
+            "h" | "left" => move_char_left,
+            "j" | "down" => move_line_down,
+            "k" | "up" => move_line_up,
+            "l" | "right" => move_char_right,
+
+            "t" => find_till_char,
+            "f" => find_next_char,
+            "T" => till_prev_char,
+            "F" => find_prev_char,
+            "r" => replace,
+            "R" => replace_with_yanked,
+
+            "~" => switch_case,
+            "`" => switch_to_lowercase,
+            "A-`" => switch_to_uppercase,
+
+            "home" => goto_line_start,
+            "end" => goto_line_end,
+
+            "w" => move_next_word_start,
+            "b" => move_prev_word_start,
+            "e" => move_next_word_end,
+
+            "W" => move_next_long_word_start,
+            "B" => move_prev_long_word_start,
+            "E" => move_next_long_word_end,
+
+            "v" => select_mode,
+            "g" => { "Goto mode"
```

Comment:
Can you please reuse the old titles that you removed rather than creating new ones? Like the following.
```suggestion
            /// goto
            ///
            /// When specified with a count, it will go to that line without entering the mode.
            "g" => {
```
The extra description will only be shown in the rustdoc but the "goto" word will be shown in both rustdoc and infobox.

Reply:
Reusing the old titles seem better since they are shorter 👍🏾. But specifying it as a doc comment and then having an additional description would be hard to parse since the macro is recursive (also writing descriptions for rustdoc seem redundant since they are not user facing and only developers will see it)

Reply:
Maybe I will take a look at this later myself.

Reply:
But can you please keep them as normal comments here?

Reply:
I changed the descriptions themselves in ad7e412